src/test/java/org/jabref/logic/layout/LayoutEntryTest.java

```diff
@@ -95,4 +101,34 @@ void parseMethodCalls() {
         assertEquals("test", (LayoutEntry.parseMethodsCalls("bla(test),foo(fark)").getFirst()).get(1));
         assertEquals("fark", (LayoutEntry.parseMethodsCalls("bla(test),foo(fark)").get(1)).get(1));
     }
+
+    private void assertUnsupportedOperation(int type, List<StringInt> parsedEntries, BibDatabaseContext context) {
+        LayoutEntry layoutEntry = new LayoutEntry(parsedEntries, type, null, null, null);
+        assertThrows(UnsupportedOperationException.class, () -> layoutEntry.doLayout(context, StandardCharsets.UTF_8));
+    }
+
+    private void assertLayoutResult(int type, String expected, List<StringInt> parsedEntries, BibDatabaseContext context) {
+        LayoutEntry layoutEntry = new LayoutEntry(parsedEntries, type, null, null, null);
+        assertEquals(expected, layoutEntry.doLayout(context, StandardCharsets.UTF_8));
+    }
+
+    @Test
+    void doLayoutTest() {
+        List<StringInt> parsedEntries = List.of(new StringInt("place_holder", 0),
+                new StringInt("testString", 0));
+        BibDatabaseContext bibDatabaseContext = new BibDatabaseContext(new BibDatabase(), new MetaData(), null);
+
+        List<Integer> unsupportedOperationTypes = List.of(2, 3, 4, 6, 7);
+        unsupportedOperationTypes.forEach(type -> assertUnsupportedOperation(type, parsedEntries, bibDatabaseContext));
+
+        // test IS_LAYOUT_TEXT
+        assertLayoutResult(1, "testString", parsedEntries, bibDatabaseContext);
+        // test IS_OPTION_FIELD with empty dataBase in BibDatabaseContext
+        assertLayoutResult(5, "testString", parsedEntries, bibDatabaseContext);
+        // test IS_ENCODING_NAME
+        assertLayoutResult(8, StandardCharsets.UTF_8.toString(), parsedEntries, bibDatabaseContext);
+        // test IS_FILENAME and IS_FILEPATH with empty path
+        assertLayoutResult(9, "", parsedEntries, bibDatabaseContext);
+        assertLayoutResult(10, "", parsedEntries, bibDatabaseContext);
```

Comment:
Looks like `@ParameterizedTest` should be used. --> There should be only one assertion per test case. (More readings at https://devdocs.jabref.org/code-howtos/testing.html#general-hints-on-tests)

Reply:
My account was flagged by automated systems while I didn't do anything violate the terms of service, finally I get my account back by sending ticket. Sorry for left this PR behind about 15 days.

Reply:
Welcome back! - I started to miss you - good to hear that you are back! Looking forward for more code-quality pull requests. Especially many tests should be converted to `@ParameterizedTests`. - I can send you pointers if I come along of some of these.